task2/src/services/users.ts

```diff
@@ -0,0 +1,101 @@
+import { Request, Response } from 'express';
+import { v4 as uuidv4 } from 'uuid';
+
+import { User } from '../types/user';
+
+const users: User[] = [];
+
+export const findById = (req: Request, res: Response) => {
+    const { id } = req.params;
+    const user = users.find((userItem) => userItem.id === id);
+
+    if (!user) {
+        res.status(404).json({
+            error: 'User not exists'
+        });
+    } else {
+        res.status(200).json(user);
+    }
+};
+
+export const findAll = (req: Request, res: Response) => {
+    const { limit = users.length, loginSubstring = '' }: any = req.query;
+
+    const result: User[] = users.filter((user, index) => user.login.includes(loginSubstring)  && index < limit)
```

Comment:
I believe that we should get all the filtered results and only after sorting them cut the requested limit

Reply:
Ok, I fixed this.